### PR TITLE
Use compressed tx size also when selecting txs from block creation

### PR DESCRIPTION
**Pull request by fab-10**
@julien-marchand this PR uses the compressed tx size also when calculating the profitability during block creation, so it is aligned with the `linea_estimateGas` endpoint, is that correct?

FYI: @macfarla @pinges check `private double getCompressedTxSize(final Transaction transaction) {` for the integration of the `gnarkCompression`

### CODE REVIEW

**Comment by fab-10** [comment]
relates to https://github.com/Consensys/protocol-misc/issues/860

**Comment by julien-marchand** [comment]
> this PR uses the compressed tx size also when calculating the profitability during block creation, so it is aligned with the linea_estimateGas endpoint, is that correct?

Yes, txpool acceptance criteria (using `eth_sendRawTransaction` or `p2p`), `linea_estimateGas` and block creation should all use the same compression mechanism. It is just the margin config that should change.

From what I understand, this PR aligns the behavior, and the PR to transform this ratio (`size/5`) into the actual compression is pending

File: arithmetization/src/main/java/net/consensys/linea/bl/TransactionProfitabilityCalculator.java [hunk 1]
```diff
@@ -55,8 +54,8 @@ public Wei profitablePriorityFeePerGas(
         .addArgument(conf.estimateGasMinMargin())
         .addArgument(conf.gasPriceRatio())
         .addArgument(conf.verificationGasCost())
-        .addArgument(txSize)
-        .addArgument(compressedSerializedSize)
+        .addArgument(transaction::getSize)
```
Comment (gauravahuja) [inline arithmetization/src/main/java/net/consensys/linea/bl/TransactionProfitabilityCalculator.java:57]: This value will be different from the earlier logged `txSize` which also added `conf.adjustTxSize()`. Is that expected?

**Review by gauravahuja — APPROVED** [review]
lgtm, just the one question.

File: arithmetization/src/main/java/net/consensys/linea/bl/TransactionProfitabilityCalculator.java (hunk 1 above)
Comment (fab-10) [reply arithmetization/src/main/java/net/consensys/linea/bl/TransactionProfitabilityCalculator.java:62]: good point, I have aligned the fields in the logs

**Pull request merged**